pkg/apis/monitoring/v1alpha1/types.go

```diff
@@ -58,8 +58,11 @@ type MonitoringStackSpec struct {
 	LogLevel LogLevel `json:"logLevel,omitempty"`
 
 	// Label selector for Monitoring Stack Resources.
+	// Set to the empty LabelSelector to monitoring everything.
+	// Set to null to disable service discovery.
 	// +optional
-	ResourceSelector *metav1.LabelSelector `json:"resourceSelector,omitempty"`
+	// +nullable
```

Comment:
Shouldn't we set the default to have service discovery enabled? i.e. - `{}` ?

Reply:
Setting a default annotation here has no effect on the generated manifest. I haven't looked into the depth of the OpenAPI code but I suspect `default` and `nullable` doesn't go well together.